smart/tests/test_map_processing.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+import pytest
+
+import astropy.units as u
+
+from sunpy.map import Map, all_coordinates_from_map, coordinate_is_on_solar_disk
+from sunpy.map.mapbase import GenericMap
+
+from smart.map_processing import cosine_correction, get_cosine_correction, map_threshold
+
+
+@pytest.fixture
+def hmi_nrt():
+    return "https://solmon.dias.ie/data/2024/06/06/HMI/fits/hmi.m_720s_nrt.20240606_230000_TAI.3.magnetogram.fits"
+
+
+@pytest.fixture
+def mag_map_sample():
+    return map_threshold(Map(hmi_nrt()))
+
+
+@pytest.fixture
+def create_fake_map(value, shape=((4098, 4098))):
+    fake_data = np.ones(shape) * value
+    return Map(fake_data, mag_map_sample().meta)
+
+
+def test_map_threshold(im_map):
+    processed_map = map_threshold(im_map)
+
+    assert isinstance(processed_map, GenericMap), "Result is not a SunPy Map."
+
+    coordinates = all_coordinates_from_map(processed_map)
+    on_solar_disk = coordinate_is_on_solar_disk(coordinates)
+    assert np.all(np.isnan(processed_map.data[~on_solar_disk])), "Off-disk NaN values not set correctly."
+
+
+def test_get_cosine_correction_shape(im_map):
+    cos_cor, d_angular, off_limb = get_cosine_correction(im_map)
+    assert cos_cor.shape == im_map.data.shape, "cos_cor shape != im_map.data.shape"
+    assert d_angular.shape == im_map.data.shape, "d_angular shape != im_map.data.shape"
+    assert off_limb.shape == im_map.data.shape, "off_limb shape != im_map.data.shape"
+
+
+def test_get_cosine_correction_limits(im_map):
```

Comment:
what you really want to test here is the actual correction. So would be able to generate some fake data assuming radial field and correct and get back constant field

```
coordinates = all_coordinates_from_map(im_map)
los_radial = cos(coordinates.Tx).cos(coordinates.Ty)
in_map.data[:,:] = los_radial
```
You'd need to double check the maths there but the idea being

```
test_corrected_data = cosine_correction(fake_map, fake_cosmap)
assert test_corrected_data == 1.0
```